src/vs/workbench/services/hover/browser/hover.ts

```diff
@@ -58,6 +58,13 @@ export interface IHoverOptions {
 	 */
 	target: IHoverTarget | HTMLElement;
 
+	/**
+	 * An ID to associate with the hover to be used as an equality check. Normally when calling
+	 * {@link IHoverService.showHover} the options object itself is used to determine if the hover
+	 * is the same one that is already showing, when this is set, the ID will be used instead.
+	 */
+	id?: number;
```

Comment:
Coming back to this, I feel a string would be better.

Reply:
https://github.com/microsoft/vscode/pull/174467